src/index.ts

```diff
@@ -131,16 +147,49 @@ export function createSelectorCreator<
     // TODO Rethink this change, or find a way to expose more options?
     const selector = defaultMemoize(function dependenciesChecker() {
       const params = []
-      const length = dependencies.length
+      const { length } = dependencies
 
       for (let i = 0; i < length; i++) {
         // apply arguments instead of spreading and mutate a local list of params for performance.
         // @ts-ignore
         params.push(dependencies[i].apply(null, arguments))
       }
 
+      const finalStabilityCheck = inputStabilityCheck ?? globalStabilityCheck
+
+      if (
+        process.env.NODE_ENV !== 'production' &&
+        (finalStabilityCheck === true ||
+          (finalStabilityCheck === 'once' && firstRun))
+      ) {
+        const paramsCopy = []
+
+        for (let i = 0; i < length; i++) {
+          // make a second copy of the params, to check if we got the same results
+          // @ts-ignore
+          paramsCopy.push(dependencies[i].apply(null, arguments))
+        }
+
+        // if the memoize method thinks the parameters are equal, these *should* be the same reference
+        const equal =
+          makeAnObject.apply(null, params) ===
+          makeAnObject.apply(null, paramsCopy)
+        if (!equal) {
+          // do we want to log more information about the selector?
```

Comment:
:speech_balloon: ~~What happens if we reference `selector.name` here?  Does it end up as `"selector"`, or whatever we assigned it to?~~

Scratch that, it comes back as `"memoized"`, because `defaultMemoize.ts` has `function memoized()`.

Mmm... We could at least log out the original inputs and both sets of extracted values?

Reply:
added 
```ts
{
  arguments,
  firstInputs: params,
  secondInputs: paramsCopy
}
```
open to better names 🙂